controllers/resource/template.go

```diff
@@ -163,6 +163,22 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 	if !ok {
 		return nil, fmt.Errorf("unable to retrieve password from secret")
 	}
+	usernameCSIBytes, ok := credSecret.Data["usernameCSI"]
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieve usernameCSI from secret")
+	}
+	passwordCSIBytes, ok := credSecret.Data["passwordCSI"]
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieve passwordCSI from secret")
+	}
+	usernameCPBytes, ok := credSecret.Data["usernameCP"]
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieve usernameCP from secret")
+	}
+	passwordCPBytes, ok := credSecret.Data["passwordCP"]
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieve passwordCP from secret")
+	}
```

Comment:
I tried moving the secret objects we discussed from `template-cp.yaml` into `secrets.yaml` but cluster creation failed. They need to be there together with the cluster config object it seems.

I did find thoug that we are already using the 2nd approach we discussed of storing credentials in a secret and then reading them from the controller when reconciling.

Unfortunately they are reading a global secret file.

I need to wrap this up to get back to cluster autoscaler work, but I think we should raise that the vSphere provider currently only really supports a single set of credentials.

I can open a ticket to audit the codebase a fix it up to support multiple credentials by cluster in future if we want.